### ISSUE TITLE
State change not called

### ISSUE THREAD

**Issue by psx88** [saeid/TestLinphone#1]
Hello,
thank you for this linphone startup in swift.

Sadly the callbacks (e.g. the registration) aren't working.
For example "print("Hey!!! LinphoneRegistrationOk")" isn't called. Although the debug message shows that the registration is working.

It seems that it's broken since the upgrade from static libraries to frameworks.

Is this working for you?

Thanks

**Comment by saeid** [comment]
Hi @psx88 ! 
sorry for late response. I checked the sdk before and it was working with the whole process. 
i will test again and check if they released an update on the main sdk. 
if this, i will compile the new sdk and update the repo. 
if you already fixed the issue, please submit a pull request! Thanks :)

**Comment by psx88** [comment]
Hello @saeid,

I ended up building a fresh xcode project and building the sdk directly from the git sources. Unfortunately that didn't fix the problem. I found some older source codes (which are working fine) and compared them to your code. The problem was with deprecated code. I'm not sure which line actually caused the trouble (currently not in front of the code) but I think it was the creation of the linphone core. Kept the deprecated line and the whole thing works like a charm!
